### ISSUE TITLE
Runtime Error and Halt after making Campaign changes

### ISSUE THREAD

**Issue by mjalt96** [gophish/gophish#1318]
What version of Gophish are you using?: 0.7.1

Brief description of the issue: Runtime error and program halting.

What are you expecting to see happen? : That program should keep running.

What are you seeing happen?

Every time I try to create a new campain, delete or stop an existing one I get a runtime error and the program terminates.

The error I am showing happened after deleting a campain from a dashbord but also happens when I delete from campains or try to launch a new one.

It was working perfectly before.

Please provide any terminal output that may be relevant below:

![captura de ecra 86](https://user-images.githubusercontent.com/27953395/49814349-5c7ab200-fd61-11e8-9f5c-759bd3bf8847.png)

Please provide as many steps as you can to reproduce the problem:
 * Step 1 - I start gophish server on Raspberry PI using: "sudo ./gophish"
 * Step 2 - I login from a browser on another machine via public address using an open port.
 * Step 3 - On "Dashboard" I delete my running campain.
 * Step 4 - Runtime error (posted here).

**Comment by jordan-wright** [comment]
Thanks for reporting this! I think this is a bug where the campaign is deleted, but emails are still scheduled to be sent.

When this happens, Gophish can't generate the emails anymore, since the campaign information is gone. At least, that's my guess of what's happening here.

I'll file this as a bug and will get it fixed asap.

**Comment by jordan-wright** [comment]
Hey @mjalt96,

I've made a change in c14be36 that should make this more resilient. I've marked it as fixed, though just know it's not 100%. For example, let's say a campaign has already been launched and a batch of emails are actively being sent, then unfortunately there isn't a way to stop them at the moment. This would be a bigger fix later on.

For now, if a campaign is scheduled for the future, and the campaign is deleted before it was set to launch, then now the maillogs are deleted and the background worker won't pick them up.
